### DISCUSSION TITLE
Removing a repository from search index

### DISCUSSION THREAD

**Discussion by nirmals-workspace** [community/community#60754] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello guys... Hope you guys are doing good... I need to stop an entire repo from being listed in google search index... Is there any possible way to do this? I tried adding the repo to Google Search Console's property, but it seems impossible.... Kindly help me on this

**Comment by Singulariity** [comment]
According to https://github.com/robots.txt, as long as you don't use the master branch you should be fine.

**Reply by nirmals-workspace** [reply]
Hello @Singulariity! Thanks for responding... Will it take time if I rename the branch now? If yes how much time will it take approximately?

**Reply by Singulariity** [reply]
I don't know exactly how long it will take, but yes, it will take some time to update.

**Comment by nirmals-workspace** [comment]
But I had repos with `main` as default branch... Even then, the repo has got displayed in search engine.... Do you have any idea about it?

**Comment by nirav-gajera** [comment]
1. Create a robots.txt file: In the root directory of your repository, create a file named robots.txt. You can do this locally and push the file to your repository. 
`User-agent: *
Disallow: /
`

-Commit and push the robots.txt file: Once you have added the robots.txt file locally, commit the file to your repository and push the changes to GitHub. This will make the file available on the web.

Verify the robots.txt file: To ensure that the robots.txt file is correctly implemented, you can access https://github.com/<username>/<repository>/robots.txt in your browser. Replace <username> with your GitHub username and <repository> with the name of your repository. You should see the contents of the robots.txt file displayed in the browser.

**Comment by nirmals-workspace** [comment]
Hello @nirav-gajera! Thanks for your input... I will try it out....

Everything sounds clear except for the thing below.... What do you mean by this? Can you make it a bit clear?

> Replace with your GitHub username and with the name of your repository. You should see the contents of the robots.txt file displayed in the browser.

**Comment by nirmals-workspace** [comment]
Also, the robots.txt should contain just `User-agent: * Disallow: / ` or I should replace this thing in the actual robots.txt file?

**Comment by nirav-gajera** [comment]
if your GitHub username is "Nirmal-Data-Scientist" and your repository name is "myrepository," the URL to access the robots.txt file would be:

https://github.com/Nirmal-Data-Scientist/myrepository/robots.txt

By visiting this URL in your browser, you should be able to see the contents of the robots.txt file displayed. 

and  he robots.txt file should contain only the line "User-agent: * Disallow: /". You should replace the entire content of the robots.txt file with this line. The robots.txt file is a plain text file that follows a specific syntax. In this case, you are using the "User-agent" directive to specify all user agents (represented by "*") and the "Disallow" directive to instruct them to not access any part of your website ("/").

**Reply by nirmals-workspace** [reply]
I have done things as you described. I have robots.txt in remote repo as well... But, now it is not displaying the robots.txt's content when I do something like https://github.com/Nirmal-Data-Scientist/myrepository/robots.txt...

For example, try with this one https://github.com/Nirmal-Data-Scientist/Term-Deposit-Subscription-Prediction/robots.txt

**Comment by nirav-gajera** [comment]
it may take some time for the changes to be reflected and for the updated robots.txt file to be visible.

If you have recently committed the robots.txt file, I recommend waiting for some time and then accessing the URL again to see if the updated content is displayed. If it still shows "Not found" after waiting for a while, ensure that the robots.txt file is correctly committed and pushed to your repository. Double-check the file path and its contents.

If the issue persists, you can also try clearing your browser cache or accessing the URL from a different browser or device to rule out any local caching issues.


and  write  Disallow: / in new line  
-
User-agent: *
Disallow: /

**Comment by nirmals-workspace** [comment]
I have recently changed my default branch names from `main` in some repos and `master` in some repos to `default` in almost all repos... Could this be a reason?
